vcd/resource_vcd_ip_space_test.go

```diff
@@ -0,0 +1,754 @@
+					resource.TestCheckResourceAttrSet("vcd_ip_space.space1", "id"),
+					resource.TestCheckResourceAttr("vcd_ip_space.space1", "route_advertisement_enabled", "false"),
+					resource.TestCheckResourceAttr("vcd_ip_space.space1", "type", "PUBLIC"),
+					resource.TestCheckResourceAttr("vcd_ip_space.space1", "internal_scope.#", "3"),
+					resource.TestCheckTypeSetElemAttr("vcd_ip_space.space1", "internal_scope.*", "192.168.1.0/24"),
+					resource.TestCheckTypeSetElemAttr("vcd_ip_space.space1", "internal_scope.*", "10.10.10.0/24"),
+					resource.TestCheckTypeSetElemAttr("vcd_ip_space.space1", "internal_scope.*", "11.11.11.0/24"),
+					resource.TestCheckResourceAttr("vcd_ip_space.space1", "external_scope", ""),
+				),
+			},
+			{
+				ResourceName:      "vcd_ip_space.space1",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateId:     t.Name(),
+			},
+			{
+				Config: configText6DS,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("vcd_ip_space.space1", "id"),
+					resourceFieldsEqual("data.vcd_ip_space.space1", "vcd_ip_space.space1", nil),
+				),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccVcdIpSpacePublicStep1 = `
+resource "vcd_ip_space" "space1" {
+  name = "{{.TestName}}"
+  type = "PUBLIC"
+
+  internal_scope = ["192.168.1.0/24"]
+
+  route_advertisement_enabled = false
+}
+`
+
+const testAccVcdIpSpacePublicStep2 = `
+resource "vcd_ip_space" "space1" {
+  name = "{{.TestName}}"
+  type = "PUBLIC"
+
+  internal_scope = ["192.168.1.0/24"]
+  external_scope = "8.8.8.0/23"
+
+  route_advertisement_enabled = true
+}
+`
+
+const testAccVcdIpSpacePublicStep3 = `
+resource "vcd_ip_space" "space1" {
+  name        = "{{.TestName}}"
+  description = "added description"
+  type        = "PUBLIC"
+
+  internal_scope = ["192.168.1.0/24","10.10.10.0/24", "11.11.11.0/24"]
+
+  route_advertisement_enabled = false
+}
+`
+
+const testAccVcdIpSpacePublicStep4 = `
+resource "vcd_ip_space" "space1" {
+  name        = "{{.TestName}}"
+  description = "added description"
+  type        = "PUBLIC"
+
+  internal_scope = ["192.168.1.0/24","10.10.10.0/24", "11.11.11.0/24"]
+
+  route_advertisement_enabled = false
+
+  ip_prefix {
+	default_quota = 2
+
+	prefix {
+		first_ip = "192.168.1.100"
+		prefix_length = 30
+		prefix_count = 4
+	}
+  }
+
+  ip_prefix {
+	default_quota = -1
+
+	prefix {
+		first_ip = "10.10.10.96"
+		prefix_length = 29
+		prefix_count = 4
+	}
+  }
+}
+`
+
+const testAccVcdIpSpacePublicStep5 = `
+resource "vcd_ip_space" "space1" {
+  name        = "{{.TestName}}"
+  description = "added description"
+  type        = "PUBLIC"
+
+  internal_scope = ["192.168.1.0/24","10.10.10.0/24", "11.11.11.0/24"]
+
+  route_advertisement_enabled = false
+
+  ip_prefix {
+	default_quota = 2
+
+	prefix {
+		first_ip = "192.168.1.100"
+		prefix_length = 30
+		prefix_count = 4
+	}
+
+	prefix {
+		first_ip = "192.168.1.200"
+		prefix_length = 30
+		prefix_count = 4
```

Comment:
indentation as in previous comments

Reply:
I have hopefully fixed all the indentation problems (done `terraform fmt` on all test snippets)